meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClientCache.java

```diff
@@ -43,11 +44,10 @@ public class ZkMetaClientCache<T> extends ZkMetaClient<T> implements MetaClientC
     private ChildChangeListener _eventListener;
     private boolean _cacheData;
     private boolean _cacheChildren;
-    private boolean _lazyCaching;
     private static final Logger LOG = LoggerFactory.getLogger(ZkMetaClientCache.class);
     private  ZkClient _cacheClient;
-
     private ExecutorService executor;
+    private final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
Since only we need one count down, see if we could use conditional variable.
Also better naming please. :D

Reply:
Will address latch in future pr to use conditional variable!